src/util/TokenService.js

```diff
@@ -59,27 +59,24 @@ class TokenService {
                 typeof tokenOrTokenFunction !== 'function') ||
             (!id.startsWith(TYPED_ID_FOLDER_PREFIX) && !id.startsWith(TYPED_ID_FILE_PREFIX))
         ) {
-            return Promise.reject(error);
+            throw error;
         }
 
         // Token is a simple string or null or undefined
         if (!tokenOrTokenFunction || typeof tokenOrTokenFunction === 'string') {
-            return Promise.resolve(tokenOrTokenFunction);
+            return tokenOrTokenFunction;
         }
 
         // Token is a function which returns a promise
-        // that on resolution returns an id to token map.
-        return new Promise(async (resolve: Function, reject: Function) => {
-            // $FlowFixMe Already checked null above
-            const token = await tokenOrTokenFunction(id);
-            if (!token || typeof token === 'string') {
-                resolve(token);
-            } else if (typeof token === 'object') {
-                resolve(token[id]);
-            } else {
-                reject(error);
-            }
-        });
+        // that on resolution returns a token.
+        const token = await tokenOrTokenFunction(id);
+        if (!token || typeof token === 'string') {
```

Comment:
Any reason why this is split between two cases that return the same thing? If you want to keep it split, it might make more sense to have the undefined/null case, and then a case where the token exists, either being a string or a well formed object.

Reply:
I'll combine them